bin/sequencer_report

```diff
@@ -0,0 +1,265 @@
+#!/usr/bin/env python3
+# pylint: disable=bad-indentation
+# pylint: disable=line-too-long
+""" Generate markdown report from sequencer results """
+import argparse
+import copy
+import json
+import os
+import sys
+from dataclasses import dataclass
+
+import jinja2
+
+TEMPLATE_PATH = "etc/sequencer_template.md"
+OUT_FILE = "results.md"
+
+# minimum set of schemas (incase device doesn't publish given message typ)
+SCHEMAS = ["state_update", "event_pointset", "event_system"]
+
+# stages for column headers
+STAGES = ["stable", "beta", "preview", "alpha"]
+
+# stages to consider for a pass
+STAGES_FOR_PASS = ["stable", "beta"]
+DEFAULT_SCORE = 1
+
+
+class TemplateHelper:
+    """Collection of helper functions made available within Jinja template"""
+
+    @staticmethod
+    def pretty_dict(thing: dict):
+        """Pretty print dictionary, e.g. key1: value1, key2: value2"""
+        if not isinstance(thing, dict):
+            return ""
+        return ", ".join([": ".join([k, v]) for k, v in thing.items()])
+
+    @staticmethod
+    def md_table_header(cols: int):
+        """Generate md table headers ( |---|--...) based on no. of cols"""
+        return " --- ".join(["|"] * (cols + 1))
+
+    @staticmethod
+    def result_icon(result):
+        """Textural representation of Boolean or None values"""
+        if result is True:
+            return "✓"
+        if result is False:
+            return "✕"
+        return "-"
+
+
+@dataclass
+class FeatureScore:
+    """Container for scored points and total points for a feature and stage"""
+
+    scored: int = 0
+    total: int = 0
+    stage: str = ""
+
+    def add(self, result, score):
+        self.total += score
+        if result == "pass":
+            self.scored += score
+
+    def has(self):
+        """Did the sequencer results have this feature & stage combination?"""
+        return self.total > 0
+
+    def passed(self):
```

Comment:
this is logic (not just rendering) -- we should consider moving it upstream to the java code, otherwise it might get out of sync with some other code that does the same thing!  Maybe not necessary now, but at least add a TODO that says something to that effect.  E.g., what happens if we decide (pass is getting 80%?)... :-P

Reply:
True - note added